src/test/java/org/jabref/model/strings/StringUtilTest.java

```diff
@@ -367,6 +368,28 @@ void capitalizeFirst() {
         assertEquals("Aa", StringUtil.capitalizeFirst("AA"));
     }
 
+    @ParameterizedTest
+    @CsvSource({
+            "''                             ,                                     , -5",
+            "''                             ,                                     ,  0",
+            "''                             ,                                     ,  3",
+            "''                             ,                                     ,  5",
+            "''                             ,                                     , 10",
+            "''                             , thisisatestfile.pdf                 , -5",
+            "''                             , thisisatestfile.pdf                 ,  0",
+            "...                            , thisisatestfile.pdf                 ,  3",
+            "...                            , thisisatestfile.pdf                 ,  5",
+            "th...e.pdf                     , thisisatestfile.pdf                 , 10",
+            "thisisatestfile.pdf            , thisisatestfile.pdf                 , 20",
+            "...                            , longfilename.extremelylongextension ,  5",
+            "...                            , longfilename.extremelylongextension , 10",
+            "...                            , longfilename.extremelylongextension , 20",
+            "lo...me.extremelylongextension , longfilename.extremelylongextension , 30",
+    })
+    void shortenFileName(String expected, String fileName, int maxLength) {
+        assertEquals(expected, StringUtil.shortenFileName(fileName, maxLength));
+    }
+
```

Comment:
Potentially add a test for `null` inputs for file name?

Reply:
Definitely! Using the @CsvSource, I have already added some tests with null fileNames. In the CSV-format strings, the first field is the expected string, the second is the inputted fileName, and lastly is the maxLength
![Screenshot 2024-10-25 at 21 11 44](https://github.com/user-attachments/assets/a233d395-3169-443d-995d-6a63e18af170)

However, thanks to your suggestion, I identified another potential null case, which is the maxLength, and will also add tests for this case.

Thanks @mag-sun !